x/ibc/07-tendermint/types/client_state.go

```diff
@@ -399,9 +398,19 @@ func validateVerificationArgs(
 		return sdkerrors.Wrap(commitmenttypes.ErrInvalidProof, "proof cannot be empty")
 	}
 
+	_, ok := proof.(commitmenttypes.MerkleProof)
+	if !ok {
+		return sdkerrors.Wrapf(commitmenttypes.ErrInvalidProof, "invalid proof type %T, expected MerkleProof", proof)
+	}
```

Comment:
we need to cast the proof to the concrete type now that there are multiple concrete implementations for the `Proof` interface